src/js/components/home/ProgressSection.js

```diff
@@ -0,0 +1,144 @@
+
+class ProgressSection extends React.Component {
+  constructor(props) {
+    super(props);
+    this.animateProgess = this.animateProgess.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener('scroll', this.animateProgess);
+    this.props.getProgresses()
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('scroll', this.animateProgess);
+  }
+
+  getProgressValueAndPercentage(progressData, key, total) {
+    if (progressData) {
+      return {
+        value: progressData[key],
+        percentage: Math.round(progressData[key] / total * 100),
+        total: total
+      };
+    }
+    return { value: 0, percentage: 0, total: total };
+  }
+
+  animateProgess(event) {
+    document.querySelectorAll('.progress-circle').forEach((progressCircle, index) => {
+      const scrollTop = document.body.scrollTop;
+      const progressCircleTop = progressCircle.getBoundingClientRect().top + window.scrollY;
+      const heightToStartOfProgressCircle = progressCircleTop + progressCircle.clientHeight - window.innerHeight;
+
+      if (scrollTop < heightToStartOfProgressCircle) {
+        progressCircle.setAttribute('data-progress', 0);
+      } else {
+        if (progressCircle.getAttribute('data-progress') !== progressCircle.getAttribute('data-goal')) {
+          progressCircle.setAttribute('data-progress', progressCircle.getAttribute('data-goal'));
+        }
+      }
+    }, this);
+  }
+
+  generateProgressCircle(data, goal, color) {
+    let progressStat;
+    const { hasBeenFetched } = this.props.progress;
+    if (!hasBeenFetched) {
+      progressStat = <InlineLoader/>
+    } else {
+      progressStat = (
+        <span>
+          {data.value}/<span>{data.total}</span>
+        </span>
+      );
+    }
+
+    return (
+      <div className="col-md-4">
+        <div className={`progress-circle progress-circle-${color}`} data-goal={data.percentage} data-progress="0">
+          <div className="circle">
+              <div className="full progress-circle-slice">
+                  <div className="progress-circle-fill" />
+              </div>
+              <div className="progress-circle-slice">
+                  <div className="progress-circle-fill" />
+                  <div className="progress-circle-fill progress-circle-bar" />
+              </div>
+          </div>
+          <div className="progress-circle-overlay">
+            <span className="progress-circle-percent">
+              {progressStat}
+              <p className="progress-circle-caption">{goal}</p>
+            </span>
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  render() {
+    const { progressData } = this.props.progress;
+    const countriesData = this.getProgressValueAndPercentage(progressData, 'countries', 2);
+    const schoolsData = this.getProgressValueAndPercentage(progressData, 'schools', 100);
+    const studentsData = this.getProgressValueAndPercentage(progressData, 'students', 1000);
```

Comment:
Might re-calculating these values on every render not get expensive? How often would progress data change? Might we consider setting this as state returned by an init function in the constructor?